test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -147,7 +151,9 @@ func setup(t *testing.T) (*httptest.Server, *garbagecollector.GarbageCollector,
 	metaOnlyClientPool := dynamic.NewClientPool(config, api.Registry.RESTMapper(), dynamic.LegacyAPIPathResolverFunc)
 	config.ContentConfig.NegotiatedSerializer = nil
 	clientPool := dynamic.NewClientPool(config, api.Registry.RESTMapper(), dynamic.LegacyAPIPathResolverFunc)
-	gc, err := garbagecollector.NewGarbageCollector(metaOnlyClientPool, clientPool, api.Registry.RESTMapper(), deletableGroupVersionResources)
+	cachedDiscoveryClient := kubectlutil.NewCachedDiscoveryClient(discovery.NewDiscoveryClientForConfigOrDie(config), os.TempDir(), time.Duration(math.MaxInt64))
```

Comment:
Factoring out `getCachedFile` and `writeCachedFile` into an interface `CacheBackend` with a memory based implementation should be easy (just using a `map[string]runtime.Object` as data structure).